src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -34,5 +43,1651 @@ public static int TrailingZeroCount(int matches)
+        {
+            int oldValue;
+            int newValue;
+            bool updated;
+            do
+            {
+                oldValue = flags;
+                newValue = oldValue | mask;
+                updated = oldValue == Interlocked.CompareExchange(ref flags, newValue, oldValue);
+            }
+            while (!updated);
+        }
+
+        #endregion
+
+        #region ClearBit (Legacy)
+
+        // https://raw.githubusercontent.com/dotnet/roslyn/367e08d8f9af968584d5bab84756eceda1587bd9/src/Workspaces/Core/Portable/Utilities/CompilerUtilities/ImmutableHashMap.cs
+        private static uint RemoveBit_Hash(int position, uint bits)
+        {
+            Debug.Assert(0 != (bits & (1u << position)));
+            return bits & ~(1u << position);
+        }
+
+        // https://raw.githubusercontent.com/dotnet/corefx/bd414c68872c4e4c6e8b1a585675a8383b3a9555/src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Utils.cs
+        internal static void ClearBit_AD(ref int value, uint bitmask)
+        {
+            value = (int)(((uint)value) & ((uint)(~bitmask)));
+        }
+
+        // https://raw.githubusercontent.com/dotnet/iot/93f2bd3f2a4d64528ca97a8da09fe0bfe42d648f/src/devices/Mcp23xxx/Mcp23xxx.cs
+        internal static void ClearBit_Mcp(ref byte data, int bitNumber)
+        {
+            ValidateBitNumber(bitNumber);
+            data &= (byte)~(1 << bitNumber);
+        }
+
+        #endregion
+
+        #region ReadBit (Legacy)
+
+        // https://raw.githubusercontent.com/dotnet/iot/93f2bd3f2a4d64528ca97a8da09fe0bfe42d648f/src/devices/Mcp23xxx/Mcp23xxx.cs
+        internal static bool GetBit_Mcp(byte data, int bitNumber)
+        {
+            ValidateBitNumber(bitNumber);
+            return ((data >> bitNumber) & 1) == 1;
+        }
+
+        internal static void ValidateBitNumber(int bitNumber)
+        {
+            if (bitNumber < 0 || bitNumber > 7)
+            {
+                throw new IndexOutOfRangeException("Invalid bit index.");
+            }
+        }
+
+        // https://raw.githubusercontent.com/dotnet/wpf/2cbb1ad9759c32dc575c7537057a29ee7da2e1b2/src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Schema/Reflector.cs
+        public static bool? GetFlag(int bitMask, int bitToCheck)
+        {
+            int validBit = GetValidMask(bitToCheck);
+            if (0 != (bitMask & validBit))
+            {
+                return 0 != (bitMask & bitToCheck);
+            }
+            return null;
+        }
+
+        private static int GetValidMask(int flagMask)
+        {
+            // Make sure we're only using the low 16 bits for flags)
+            Debug.Assert((flagMask & 0xFFFF) == flagMask, "flagMask should only use lower 16 bits of int");
+            return flagMask << 16;
+        }
+
+        #endregion
+
+        // Proposed new code.
+        // Does not use hardware intrinsics yet.
+
+        #region PopCount
```

Comment:
This is the start of the new (suggested) code.
Note that it doesn't use intrinsics yet.

Reply:
Are you indicating that these ones aren't currently used by any existing code in CoreFX/CoreCLR?

Reply:
No, this is the code I would like all the other methods to eventually call. It is unit-tested and benchmarked (in a separate repo). The 'legacy' duplicate methods above this will eventually be removed and the call sites will be updated to call these. Assuming the naming, etc passes review.

Reply:
Unless you are suggesting another way of migrating the call sites - I am trying to balance the need for evidence of duplication, with the need to track what has changed?

Reply:
No. Was just wanting to ensure we were only covering code that was being actively used in CoreFX/CoreCLR.

Merging all the duplicated methods into a single (without worrying too much about perf, outside whichever one looks/is micro-benched to be the fastest) is probably fine for now.

Reply:
We can revisit on improving the implementation later, as this is mostly just to see existing usages and centralizing them (as that both cleans up the existing code and gets the usage data that the API Review team wanted).

Reply:
OK, so should I remove the duplicate methods and (perhaps) just leave the url links in place now?

Reply:
Oh I think you meant the opposite. Remove the 'new' code and just keep the 'legacy' until it has been reviewed. Please confirm what you need me to do.

Reply:
Let me know if latest commit is suitable